### ISSUE TITLE
Feature compatibility verification

### ISSUE THREAD

**Issue by HeroicKatora** [obi1kenobi/cargo-semver-checks#412]
### Describe your use case

Feature flags can be used in rustc to enable and disable arbitrary chunks of code via cfg. Due to the _additive_ nature of features during the dependency resolution of cargo, however, in many cases they are supposed to behave in a compatible manner. 

For instance, consider a crate with an API for no-std and some utilities for `std` use cases. (As implied by this construction, there is usually a feature `std` and _not_ a feature `no_std` since adding the latter in a dependency can't be undone by a sibling dependency.) Concretely the crate may:

* Have an error type which implements `std::error::Error` gated behind `cfg(feature = "std")`.
* Offer a method `parse_str(_: &str)` and additional a method `parse<R: std::io::Read>` gated behind the feature flag.

Now, a consumer may access the crate `dep-cfg` via two unrelated sibling dependencies, `dep-a` and `dep-b`. A common example may be `serde_json` as a wide-spread format. Then dependency `dep-a` may require the `std` feature while the other does not. Also `dep-b` is usually only tested against the crate _without_ the feature. The expectation of additive features is now that `dep-cfg`'s interface is only _augmented_ by adding the `std` feature but it is not broken. Note that the sibling dependency `dep-a: dep-cfg` propagates the `std` feature flag to the crate compiled for `dep-b`, i.e. `dep-cfg` is only compiled _once_ with the aggregate of all enabled features. For a stable ecosystem, `dep-b` should be able to rely on its small test suite even in the presence of unknown downstream users such as `dep-a`.

This is the same problem as compatibility checks between compatible versions.

### Describe the solution you'd like

Add support for comparing explicit feature configurations.

* A new subcommand `check-features` with flags `--baseline-features` and `--features`

Then the above check may be implemented in CI as follows:

```base
cargo-semver-checks semver-checks check-features --baseline-features "" --features "std"
```

The semantics would be similar to using `--baseline-root .` in the existing subcommand check-release. It may be implemented in terms of this subcommand.

**Comment by obi1kenobi** [comment]
Thanks for the detailed writeup, much appreciated!

This is something I've also considered, and that would be awesome to have. I think the majority of lints would be usable mostly as-is, perhaps just with tweaked error messages.

Recently, I've been thinking that the subcommand nesting `cargo semver-checks check-release` is a bit too involved, so this functionality may perhaps become a separate binary that's part of the same crate rather than a subcommand, kind of like how `cargo add` and `cargo edit` were both part of `cargo-edit` before the former was upstreamed into cargo.

Long story short, the functionality you described should exist, and it's certainly within reach!

**Comment by obi1kenobi** [comment]
[v0.21 was just released](https://github.com/obi1kenobi/cargo-semver-checks/releases/tag/v0.21.0) and allows specifying the exact set of features to use, both by feature group (e.g. only default features) as well as at the level of individual features.

The functionality in your example ("check current with only feature `std` vs baseline without any features") can be achieved like so:
`cargo semver-checks check-release --only-explicit-features --current-features std` 

The [FAQ entry for this new feature](https://github.com/obi1kenobi/cargo-semver-checks#what-features-does-cargo-semver-checks-enable-in-the-tested-crates) has more information.

Thanks for your patience! If you get a chance to try it, I'd love to hear how it went.
